### PR TITLE
Refactor the range specifier grammar

### PR DESCRIPTION
**Pull request by royfielding**
This refactors range-spec to simplify and reduce distinction with range unit extensions. Moving the extension grammar into range specifiers (and removing comma) reduces the overlapping rules and allows the stuff specific to byte ranges to be specified separately.

Related to #11, #196, #210, #212

### CODE REVIEW

**Comment by reschke** [comment]
This refactors things further than I expected, in that it also introduces consistency for range sets across range units. I'm not totally sure we can go that for (will think about it).

**Comment by reschke** [comment]
FWIW, <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range> appears to support Roy's proposal.

**Comment by royfielding** [comment]
The consistency for range sets is saying that if you have a comma-separated list of range specifiers then it means you want more than one range. It's hard to imagine any scenario in which some future range unit would want

    Range: unit=set,set,set

to have any other meaning. And that is clearly the intent of the original ABNF.

I note that the MDN articles use a space after each comma. Perhaps we need to test that?

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -2628,49 +2634,103 @@ Host: www.example.org
...
-  <x:ref>byte-ranges-specifier</x:ref> = <x:ref>bytes-unit</x:ref> "=" <x:ref>byte-range-set</x:ref>
-  <x:ref>byte-range-set</x:ref>  = 1#( <x:ref>byte-range-spec</x:ref> / <x:ref>suffix-byte-range-spec</x:ref> )
-  <x:ref>byte-range-spec</x:ref> = <x:ref>first-byte-pos</x:ref> "-" [ <x:ref>last-byte-pos</x:ref> ]
-  <x:ref>first-byte-pos</x:ref>  = 1*<x:ref>DIGIT</x:ref>
-  <x:ref>last-byte-pos</x:ref>   = 1*<x:ref>DIGIT</x:ref>
+<sourcecode type="abnf7230"><iref primary="true" item="Grammar" subitem="ranges-specifier"/><iref primary="true" item="Grammar" subitem="range-set"/><iref primary="true" item="Grammar" subitem="range-spec"/>
+  <x:ref>ranges-specifier</x:ref> = <x:ref>range-unit</x:ref> "=" <x:ref>range-set</x:ref>
+  <x:ref>range-set</x:ref>        = <x:ref>range-spec</x:ref> *( "," <x:ref>range-spec</x:ref>)
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml]: Perhaps it is better to leave this as

   <x:ref>range-set</x:ref>        = 1#<x:ref>range-spec</x:ref>

until we have decided #212

**Comment by reschke** [comment]
Looks good in general.

We can decouple this from #212 by going back to the list syntax for now (as proposed).

I still believe we have a (good) normative change here, in that we now say how ranges can be interpreted for units other than "bytes". This needs to be documented.

**Pull request merged**